src/browser/pages/vscode.ts

```diff
@@ -5,18 +5,54 @@ const options = getOptions()
 // TODO: Add proper types.
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-let nlsConfig: any
+export const nlsConfigElementId = "vscode-remote-nls-configuration"
+
+type NlsConfiguration = {
+  locale: string
+  availableLanguages: { [key: string]: string } | {}
+  _languagePackId?: string
+  _translationsConfigFile?: string
+  _cacheRoot?: string
+  _resolvedLanguagePackCoreLocation?: string
+  _corruptedFile?: string
+  _languagePackSupport?: boolean
+  loadBundle?: any
+}
+
+/**
+ * A helper function to get the NLS Configuration settings.
+ *
+ * This is used by VSCode for localizations (i.e. changing
+ * the display language).
+ **/
+export function getNlsConfiguration(document: Document) {
+  const nlsConfigElement = document?.getElementById(nlsConfigElementId)
+  const nlsConfig = nlsConfigElement?.getAttribute("data-settings")
+
+  if (!document || !nlsConfigElement || !nlsConfig) {
+    return undefined
+  }
+
+  try {
+    return JSON.parse(nlsConfig) as NlsConfiguration
+  } catch (error) {
+    console.error("[vscode] Failed to read or parse the NLS configuration.", error)
+    return undefined
+  }
+}
+
+let nlsConfig: NlsConfiguration | undefined
```

Comment:
If we have `getNlsConfiguration` throw instead of return undefined we could do something like:

```
try {
  const nlsConfig = getNlsConfiguration(document)
  if (nlsConfig._resolvedLanguagePackCoreLocation) {
    // etc
  }
  // etc
} catch (error) {
  console.error(error)
}

Reply:
Good idea! I'll refactor

Reply:
Worked brilliantly!